src/dft/descriptor.cpp

```diff
@@ -0,0 +1,149 @@
+*******************************************************************************/
+
+#include "oneapi/mkl/dft/descriptor.hpp"
+
+#ifdef ENABLE_MKLCPU_BACKEND
+#include "oneapi/mkl/dft/detail/mklcpu/onemkl_dft_mklcpu.hpp"
+#endif
+#ifdef ENABLE_MKLGPU_BACKEND
+#include "oneapi/mkl/dft/detail/mklgpu/onemkl_dft_mklgpu.hpp"
+#endif
+#include "oneapi/mkl/dft/detail/dft_loader.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+
+#ifdef BUILD_RUN
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(sycl::queue &queue) {
+    queue_ = queue;
+    pimpl_.reset(detail::create_commit(*this));
+}
+#endif
+
+#ifdef BUILD_COMP
+#ifdef ENABLE_MKLCPU_BACKEND
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(backend_selector<backend::mklcpu> selector) {
+    queue_ = selector.get_queue();
+    pimpl_.reset(mklcpu::create_commit(*this));
+}
+#endif
+
+#ifdef ENABLE_MKLGPU_BACKEND
+template <precision prec, domain dom>
+void descriptor<prec, dom>::commit(backend_selector<backend::mklgpu> selector) {
+    queue_ = selector.get_queue();
+    // pimpl_.reset(mklgpu::create_commit(*this));
+}
+#endif
+#endif
+
+// impliment error class
+template <precision prec, domain dom>
+void descriptor<prec, dom>::set_value(config_param param, ...) {
+    int err = 0;
+    va_list vl;
+    va_start(vl, param);
+    switch (param) {
+        case config_param::INPUT_STRIDES: [[fallthrough]];
+        case config_param::OUTPUT_STRIDES: {
+            int64_t *strides = va_arg(vl, int64_t *);
+            if (strides == nullptr)
+                break;
+            if (param == config_param::INPUT_STRIDES)
+                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.input_strides));
+            if (param == config_param::OUTPUT_STRIDES)
+                std::copy(strides, strides + rank_ + 1, std::back_inserter(values_.output_strides));
+        } break;
+        case config_param::FORWARD_SCALE: values_.fwd_scale = va_arg(vl, double); break;
+        case config_param::BACKWARD_SCALE: values_.bwd_scale = va_arg(vl, double); break;
+        case config_param::NUMBER_OF_TRANSFORMS:
+            values_.number_of_transforms = va_arg(vl, int64_t);
+            break;
+        case config_param::FWD_DISTANCE: values_.fwd_dist = va_arg(vl, int64_t); break;
+        case config_param::BWD_DISTANCE: values_.bwd_dist = va_arg(vl, int64_t); break;
+        case config_param::PLACEMENT: values_.placement = va_arg(vl, config_value); break;
+        case config_param::COMPLEX_STORAGE:
+            values_.complex_storage = va_arg(vl, config_value);
+            break;
+        case config_param::CONJUGATE_EVEN_STORAGE:
+            values_.conj_even_storage = va_arg(vl, config_value);
+            break;
+        default: err = 1;
+    }
+    va_end(vl);
+}
+template <precision prec, domain dom>
+descriptor<prec, dom>::descriptor(std::vector<std::int64_t> dimensions)
+        : dimensions_(dimensions),
```

Comment:
```suggestion
        : dimensions_(std::move(dimensions)),
```
To avoid another copy of the std::vector.

Just be careful to use `dimensions_` instead of `dimensions` later in the function.